fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanApplicationWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -1705,4 +1709,20 @@ private void officeSpecificLoanProductValidation(final Long productId, final Lon
         }
     }
 
+    private void validateTransactionProcessingStrategy(final LoanProduct loanProduct, final String loanRepaymentStrategy) {
+        if (loanProduct.getRepaymentStrategy().equals(LoanProductConstants.ADVANCED_PAYMENT_ALLOCATION_STRATEGY)) {
```

Comment:
This one is not necessary true.

Just because the loan product was configured to advanced payment allocation, it does not mean the loan must be as well.

The rule is:
- If the loan product WAS NOT configured for advanced payment allocation, the loan CANNOT be 'advanced payment allocation`
- If the loan product WAS configured for advanced payment allocation, the loan CAN be advanced payment allocation OR ANYTHING ELSE, but if something else is selected the loan schedule type MUST be Cumulative.
- When the loan schedule type is CUMULATIVE, then Loan and loan product repayment strategy CANNOT be "advanced payment allocation". 
- When the loan schedule type is PROGRESSIVE, then the loan and loan product repayment strategy MUST be only "advanced payment allocation".

I hope it helps!

Reply:
Done, code updated as:

```
        // PROGRESSIVE: Repayment strategy MUST be only "advanced payment allocation"
        if (loanProduct.getLoanProductRelatedDetail().getLoanScheduleType().equals(LoanScheduleType.PROGRESSIVE)) {
            if (!transactionProcessingStrategyCode.equals(LoanProductConstants.ADVANCED_PAYMENT_ALLOCATION_STRATEGY)) {
                throw new GeneralPlatformDomainRuleException(
                        "error.msg.loan.repayment.strategy.can.not.be.different.than.advanced.payment.allocation",
                        "Loan repayment strategy can not be different than Advanced Payment Allocation");
            }
            // CUMULATIVE: Repayment strategy CANNOT be "advanced payment allocation"
        } else {
            if (transactionProcessingStrategyCode.equals(LoanProductConstants.ADVANCED_PAYMENT_ALLOCATION_STRATEGY)) {
                throw new GeneralPlatformDomainRuleException(
                        "error.msg.loan.repayment.strategy.can.not.be.equal.to.advanced.payment.allocation",
                        "Loan repayment strategy can not be equal to Advanced Payment Allocation");
            }
        }

```